woocommerce/woocommerce-android
# Order Creation: product section's UI

Closes: #5511

### Description
This PR handles the product's section UI, it doesn't handle any actions yet.

I had to make some adjustments to the section's component, but I tried to keep all changes non-breaking.

### Testing instructions
1. Make sure order creation is enabled from the beta features screen on the app.
2. Apply the patch from below.
3. Open the app, click on orders.
4. Click on the FAB button.
5. Confirm that the products are displayed correctly.

*Loading the images depends on using correct product ids, and having the products cached in the app.*

### Images/gif
<img width="396" alt="Screen Shot 2021-12-23 at 11 33 51" src="https://user-images.githubusercontent.com/1657201/147227982-4cfe38c9-3995-4059-982e-9347d053edb9.png">

### Patch
```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt	(revision 485a01abb653457bed527c47f2abd8f3416d6504)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationViewModel.kt	(date 1640255589049)
@@ -16,6 +16,7 @@
 import com.woocommerce.android.viewmodel.ScopedViewModel
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.withContext
+import java.math.BigDecimal
 import javax.inject.Inject
 
 @HiltViewModel
@@ -57,6 +58,49 @@
         orderDraft = orderDraft.copy(
             currency = parameterRepository.getParameters(PARAMETERS_KEY, savedState).currencySymbol.orEmpty()
         )
+        orderDraft = orderDraft.copy(
+            items = listOf(
+                Order.Item(
+                    0L,
+                    productId = 44,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU123",
+                    quantity = 2f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                ),
+                Order.Item(
+                    0L,
+                    productId = 45,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU124",
+                    quantity = 2f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                ),
+                Order.Item(
+                    0L,
+                    productId = 47,
+                    "Test Product",
+                    price = BigDecimal.TEN,
+                    sku = "SKU125",
+                    quantity = 2f,
+                    subtotal = BigDecimal.ONE,
+                    totalTax = BigDecimal.TEN,
+                    total = BigDecimal.TEN,
+                    variationId = 0L,
+                    attributesList = emptyList()
+                )
+            )
+        )
     }
 
     fun onOrderStatusChanged(status: Order.Status) {
```
